### PR TITLE
V2 mod upgrade

### PR DESCRIPTION
**Pull request by prestonvasquez**
## Summary

Update the mod and self import paths with v2.

### CODE REVIEW

File: Makefile
Comment (prestonvasquez) [inline Makefile:29]: TODO: before merge, uncomment

File: internal/cmd/compilecheck/go.mod
Comment (prestonvasquez) [inline internal/cmd/compilecheck/go.mod:9]: TODO: before merge uncomment and update with alpha

Comment (blink1073) [reply internal/cmd/compilecheck/go.mod:9]: This is ready: https://github.com/mongodb/mongo-go-driver/actions/runs/10098917125

Comment (prestonvasquez) [reply internal/cmd/compilecheck/go.mod]: On further investigation, I think we will have to wait for a release that contains `v2` in the import path in order for the compile check to work. I suggest releasing alpha2 after this is merged, and then we can create a followup PR to fix the compile check.

@matthewdale Do you have thoughts?

Comment (prestonvasquez) [reply internal/cmd/compilecheck/go.mod]: The same goes for all tests in internal/cmd that are modules.

Comment (matthewdale) [reply internal/cmd/compilecheck/go.mod]: Doing this in a follow-up PR seems fine.

**Review by matthewdale — APPROVED** [review]
Looks good! 👍

**Pull request merged**
